### PR TITLE
Security patches

### PR DESCRIPTION
**Pull request by mmango10**
This pull request introduces a new shared validation module, backend validation hardening/security, and refactors the `admin-gear-mutate` and `admin-ops` Supabase functions to use these centralized validation utilities. The changes improve input validation consistency, reduce repetitive code, and make error handling more robust and maintainable. Additionally, comprehensive tests for the new validation logic are included.

**Validation module introduction and usage:**

* Added a new `validation.ts` module in `_shared` with reusable validators for text, enums, UUIDs, emails, integers, ISO dates, JSON objects, arrays, and object keys, along with standard regex patterns and a custom `ValidationError` class.
* Added `validation_test.ts` to thoroughly test the new validation utilities for edge cases and expected error conditions.

**Refactoring of `admin-gear-mutate`:**

* Replaced all manual input normalization and validation with calls to the new shared validation functions (`requireText`, `requireEnum`, `requireUuid`, `optionalText`, etc.), improving reliability and reducing code duplication. [[1]](diffhunk://#diff-384e7d486c6cd34bc97e15327bca86db43c11bfe6c957930d1898ad367bff926R11-R18) [[2]](diffhunk://#diff-384e7d486c6cd34bc97e15327bca86db43c11bfe6c957930d1898ad367bff926L166-R174) [[3]](diffhunk://#diff-384e7d486c6cd34bc97e15327bca86db43c11bfe6c957930d1898ad367bff926L269-R281) [[4]](diffhunk://#diff-384e7d486c6cd34bc97e15327bca86db43c11bfe6c957930d1898ad367bff926L320-R319) [[5]](diffhunk://#diff-384e7d486c6cd34bc97e15327bca86db43c11bfe6c957930d1898ad367bff926L394-R377) [[6]](diffhunk://#diff-384e7d486c6cd34bc97e15327bca86db43c11bfe6c957930d1898ad367bff926L436-R416)
* Updated error handling to return standardized error responses using `ValidationError`, ensuring consistent client feedback for invalid input.
* Updated the allowed gear statuses definition to use `as const` for stricter type safety.

**Refactoring of `admin-ops`:**

* Centralized all request action and payload validation using the new `asRecord`, `requireText`, and related utilities, eliminating ad-hoc validation logic. [[1]](diffhunk://#diff-afb1e08809fa4131c51f5416f838b379a62cd624d44f5a26538ece65275fbe6bR14-R22) [[2]](diffhunk://#diff-afb1e08809fa4131c51f5416f838b379a62cd624d44f5a26538ece65275fbe6bL348-R362)
* Updated all action checks to use a normalized action string, ensuring consistent handling and reducing risk of bugs from malformed input. [[1]](diffhunk://#diff-afb1e08809fa4131c51f5416f838b379a62cd624d44f5a26538ece65275fbe6bL582-R587) [[2]](diffhunk://#diff-afb1e08809fa4131c51f5416f838b379a62cd624d44f5a26538ece65275fbe6bL686-R691) [[3]](diffhunk://#diff-afb1e08809fa4131c51f5416f838b379a62cd624d44f5a26538ece65275fbe6bL727-R732) [[4]](diffhunk://#diff-afb1e08809fa4131c51f5416f838b379a62cd624d44f5a26538ece65275fbe6bL749-R754) [[5]](diffhunk://#diff-afb1e08809fa4131c51f5416f838b379a62cd624d44f5a26538ece65275fbe6bL839-R836) [[6]](diffhunk://#diff-afb1e08809fa4131c51f5416f838b379a62cd624d44f5a26538ece65275fbe6bL979-R976)
* Replaced custom integer normalization for tenant settings with the shared `optionalInteger` validator.
* Updated allowed gear statuses to use `as const` for improved type safety.

**Other:**

* Updated the changelog to reflect the security fixes and development update.*
* 

Summary generated by Copilot.

### CODE REVIEW

File: supabase/functions/super-auth-verify/index.ts [hunk 1]
```diff
@@ -14,6 +14,12 @@ import {
 import { registerPrivilegedStepUp } from "../_shared/privilegedStepUp.ts";
 import { isAllowedOrigin, parseAllowedOrigins } from "../_shared/cors.ts";
 import { requireTrustedEdgeIngress } from "../_shared/trustedIngress.ts";
+import {
+  optionalText,
+  requireEmail,
+  requireText,
+  ValidationError,
+} from "../_shared/validation.ts";
```
[the remark this answers is not in the document]
Comment (mmango10) [reply supabase/functions/super-auth-verify/index.ts:21]: patched

File: supabase/functions/super-auth-verify/index.ts [hunk 2]
```diff
@@ -96,10 +102,7 @@ const resolveCorsHeaders = (req: Request) => {
   return { hasOrigin, originAllowed, headers };
 };
 
-const normalizeText = (value: unknown, max = 500) => {
-  if (typeof value !== "string") return "";
-  return value.trim().slice(0, max);
-};
+const normalizeText = (value: unknown, max = 500) => optionalText(value, { maxLen: max });
```
[the remark this answers is not in the document]
Comment (mmango10) [reply supabase/functions/super-auth-verify/index.ts]: patched

File: supabase/functions/super-admin-mutate/index.ts [hunk 3]
```diff
@@ -6,6 +6,15 @@ import {
...
 import { isAllowedOrigin, parseAllowedOrigins } from "../_shared/cors.ts";
+import {
+  optionalText,
+  requireEmail,
+  requireEnum,
+  requireText,
+  requireUuid,
+  UUID_PATTERN,
+  ValidationError,
+} from "../_shared/validation.ts";
```
[the remark this answers is not in the document]
Comment (mmango10) [reply supabase/functions/super-admin-mutate/index.ts:16]: patched

File: supabase/functions/district-admin-mutate/index.ts [hunk 4]
```diff
@@ -6,6 +6,14 @@ import {
...
 } from "../_shared/privilegedStepUp.ts";
 import { isAllowedOrigin, parseAllowedOrigins } from "../_shared/cors.ts";
+import {
+  ACCESS_CODE_PATTERN,
+  optionalText,
+  requireEnum,
+  requireText,
+  requireUuid,
+  ValidationError,
+} from "../_shared/validation.ts";
```
[the remark this answers is not in the document]
Comment (mmango10) [reply supabase/functions/district-admin-mutate/index.ts:15]: patched

File: supabase/functions/admin-ops/index.ts [hunk 5]
```diff
@@ -1193,22 +1188,24 @@ serve(async (req) => {
...
-          skippedRows.push({ barcode, reason: "Field length exceeded." });
-          continue;
-        }
-        if (!ALLOWED_GEAR_STATUSES.has(statusRaw)) {
-          skippedRows.push({ barcode, reason: "Invalid status." });
+        const rowRecord = row as Record<string, unknown>;
+        let name = "";
+        let barcode = "";
+        let serial = "";
+        let statusRaw: "available" | "checked_out" | "damaged" | "lost" | "in_repair" | "retired" | "in_studio_only" = "available";
```
[the remark this answers is not in the document]
Comment (mmango10) [reply supabase/functions/admin-ops/index.ts]: patched

**Comment by mmango10** [comment]
GitHub CodeQL findings patched
